src/Function.cpp

```diff
@@ -427,8 +428,13 @@ void Function::define(const vector<string> &args, vector<Expr> values) {
 
     // Tag calls to random() with the free vars
     int tag = rand_counter++;
+    vector<VarOrRVar> free_vars;
+    free_vars.reserve(args.size());
+    for (size_t i = 0; i < args.size(); i++) {
```

Comment:
Prefer range-based for loops when the index isn't used:
```
for (const auto &arg : args) {
  free_vars.emplace_back(Var(arg));
}
```

Reply:
I propagated the changes to a few loops in proximity. Let me know if I changed too much.